### PR TITLE
(cherry pick) *: fix writing null value into not null column in write-only state.

### PR DESCRIPTION
**Pull request by jackysp**
Rewrite some parts of the original pull request. @zimulala @coocood PTAL

### CODE REVIEW

File: table/tables/tables.go [hunk 1]
```diff
@@ -473,7 +475,7 @@ func (t *Table) RowWithCols(ctx context.Context, h int64, cols []*table.Column)
 		if col == nil {
 			continue
 		}
-		if col.IsPKHandleColumn(t.meta) {
+		if col.IsPKHandleColumn(t.Meta()) {
```
Comment (coocood) [inline table/tables/tables.go]: Why not use `t.meta`?

**Comment by coocood** [comment]
LGTM

**Comment by jackysp** [comment]
/run-integration-common-test tidb-test=release-1.0 tidb-private-test=release-1.0 tikv=release-1.0
/run-common-test tidb-test=release-1.0 tidb-private-test=release-1.0 tikv=release-1.0

**Comment by jackysp** [comment]
/run-integration-common-test tidb-test=release-1.0 tidb-private-test=release-1.0 tikv=release-1.0
/run-common-test tidb-test=release-1.0 tidb-private-test=release-1.0 tikv=release-1.0

**Comment by jackysp** [comment]
PTAL @zimulala

**Pull request merged**